src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientConfigBuilder.java

```diff
@@ -35,10 +82,98 @@ public DatabaseClientConfig buildDatabaseClientConfig(Map<String, String> kafkaC
 
 		String securityContextType = kafkaConfig.get(MarkLogicSinkConfig.CONNECTION_SECURITY_CONTEXT_TYPE).toUpperCase();
 		clientConfig.setSecurityContextType(SecurityContextType.valueOf(securityContextType));
-
+		/*
+		 * v1.2.2 changes START
+		 */		
+		if ("BASIC".equals(securityContextType) ||
```

Comment:
Same thing here - toss this into a private method (similar to the existing configureSimpleSsl method - in fact, make it a protected method if that makes it easier to invoke separately as part of a unit test) with a self-documenting name like "configureCustomSslConnection". 

The goal is to avoid buildDatabaseClientConfig becoming hundreds of lines long, which makes it much more difficult to understand.

Reply:
Makes sense. Making the changes.